derive/src/utils.rs

```diff
@@ -112,13 +112,21 @@ pub fn variant_index(v: &Variant, i: usize) -> TokenStream {
 		None
 	});
 
-	// then fallback to discriminant or just index
-	index.map(|i| quote! { #i }).unwrap_or_else(|| {
-		v.discriminant
-			.as_ref()
-			.map(|(_, expr)| quote! { #expr })
-			.unwrap_or_else(|| quote! { #i })
-	})
+	// if no attribute, use the discriminant if there is one
```

Comment:
Am I confused or is this comment not the best? :D

Reply:
I had a pass over it!